### ISSUE TITLE
Missing Xcode dependency: Python module "six". Install via 'pip install six' or 'sudo easy_install six'. Could not build the application for the simulator. Error launching application on iPhone X.

### ISSUE THREAD

**Issue by Kenny1610** [flutter/flutter#17139]
I tried literally every option to be able to run my flutter project and I keep receiving this error. When I run flutter doctor -v, I get

[✓] Flutter (Channel beta, v0.2.8, on Mac OS X 10.13.4 17E199, locale en-US)
• Flutter version 0.2.8 at /Users/The_Chosen_King/Downloads/flutter
• Framework revision b397406 (4 weeks ago), 2018-04-02 13:53:20 -0700
• Engine revision c903c21
• Dart version 2.0.0-dev.43.0.flutter-52afcba357

[✓] Android toolchain - develop for Android devices (Android SDK 27.0.3)
• Android SDK at /Users/The_Chosen_King/Library/Android/sdk
• Android NDK location not configured (optional; useful for native profiling support)
• Platform android-27, build-tools 27.0.3
• Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
• Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)
• All Android licenses accepted.

[✓] iOS toolchain - develop for iOS devices (Xcode 9.3)
• Xcode at /Applications/Xcode.app/Contents/Developer
• Xcode 9.3, Build version 9E145
• ios-deploy 1.9.2
• CocoaPods version 1.5.0

[✓] Android Studio (version 3.1)
• Android Studio at /Applications/Android Studio.app/Contents
• Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)

[✓] Connected devices (2 available)
• Kendall • 2cd6bdf3effa50c19ca810fe70edbe5b3d14362c • ios • iOS 11.3
• iPhone X • 81EAB622-6A06-499C-A9BD-89BE71BF8910 • ios • iOS 11.3 (simulator)

• No issues found!

Everything is said to be working fine until I try to debug and run the the original code when you start a new flutter project and I get this:

Missing Xcode dependency: Python module "six".
Install via 'pip install six' or 'sudo easy_install six'.
Could not build the application for the simulator.
Error launching application on iPhone X.

I've tried doing the 'pip install' and the 'sudo easy_install six' and I just get: Requirement already satisfied: six in /Users/The_Chosen_King/anaconda3/lib/python3.6/site-packages (1.11.0)

I've tried uninstalling python and reinstalling using the command 'brew reinstall python@2' and I also did the same with python3. 

I don't know what I'm doing wrong if I am doing anything wrong.

**Comment by jasnig** [comment]
I also met it and could't solve!

**Comment by w548544375** [comment]
brew reinstall python@2

pip install six

i tried and it's works.

[https://stackoverflow.com/questions/49775505/missing-xcode-dependency-python-module-six](url)

**Comment by kelset** [comment]
Can confirm, the solution explained by @w548544375 worked for me too.

**Comment by Sun3** [comment]
I have tried to reinstall python with above commands but still getting errors:

```
flutter run --release
Launching lib/main.dart on Marco’s iPhone in release mode...
Automatically signing iOS for device deployment using specified development team in Xcode project: W4DESGH8Y9
Running Xcode clean...                                       1.1s
Starting Xcode build...                                          
 ├─Building Dart code...                             22.4s
 ├─Assembling Flutter resources...                    1.1s
 └─Compiling, linking and signing...                  9.7s
Xcode build done.                                           35.4s
Installing and launching...                                      
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "/usr/local/Cellar/python@2/2.7.15/Frameworks/Python.framework/Versions/2.7/lib/python2.7/copy.py", line 52, in <module>
    import weakref
  File "/usr/local/Cellar/python@2/2.7.15/Frameworks/Python.framework/Versions/2.7/lib/python2.7/weakref.py", line 14, in <module>
    from _weakref import (
ImportError: cannot import name _remove_dead_weakref
```

**Comment by foyer** [comment]
@Sun3 Same. Flutter doctor checks out with no warnings. Only building for iOS connected device

`Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "/usr/local/Cellar/python@2/2.7.15/Frameworks/Python.framework/Versions/2.7/lib/python2.7/copy.py", line 52, in <module>
    import weakref
  File "/usr/local/Cellar/python@2/2.7.15/Frameworks/Python.framework/Versions/2.7/lib/python2.7/weakref.py", line 14, in <module>
    from _weakref import (
ImportError: cannot import name _remove_dead_weakref`

It seems to run ok besides the error but would like to suppress that.

**Comment by minimaluminium** [comment]
Same here. Any solution on this?

**Comment by jobiwankanobi** [comment]
For me I had to use `/usr/local/bin/pip install six`

**Comment by cbracken** [comment]
Fixed in #19281 on `master`. It should no longer be necessary to manually install the `six` module if you have a custom Python install on your path.
